src/index.js

```diff
@@ -1,32 +1,43 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const API = 'https://rickandmortyapi.com/api/character/';
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
+
+function fetchData(url_api) {
+  return new Promise( (resolv, reject)=>{
+    let xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange =  (event) => {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200)          
+          resolv( JSON.parse( xhttp.responseText) );
+        else 
+          return reject(url_api);
+      }
+    };
+    xhttp.open('GET', url_api, true);
+    xhttp.send();
+  })
 };
 
-fetchData(API, function (error1, data1) {
-  if (error1) return console.error('Error' + ' ' + error1);
-  console.log('Primer Llamado...')
-  fetchData(API + data1.results[0].id, function (error2, data2) {
-    if (error2) return console.error(error1);
+
+fetchData( API )
+  .then( (data) =>{
+    console.log('Primer Llamado...')
+    let url = API + data.results[0].id;
+    return fetchData( url )
+  })
+  .then( (data)=>{
     console.log('Segundo Llamado...')
-    fetchData(data2.origin.url, function (error3, data3) {
-      if (error3) return console.error(error3);
-      console.log('Tercero Llamado...')
-      console.log('Personajes:' + ' ' + data1.info.count);
-      console.log('Primer Personaje:' + ' ' + data2.name);
-      console.log('Dimensión:' + ' ' + data3.dimension);
-    });
-  });
-});
\ No newline at end of file
+    console.log(data)
+    let url = data.origin.url
+    console.log( url )
+    return fetchData( url )
+  })
+  .then( (data)=>{
+    console.log('Tercero Llamado...')
+    if( data.info )
+      console.log(`Personajes: ${data.info.count}`);
+    console.log(`Primer Personaje: ${data.name}`);
+    console.log(`Dimensión: ${data.dimension}`);
+  })
+  .catch( ()=>console.log("error"))
```

Comment:
Siempre que trabajemos con un .catch, es bueno ver qué error hemos obtenido.

```suggestion
  .catch((error)=>console.log(`Error ${error}`))
```